pkg/apis/observability-ui/v1alpha1/types.go

```diff
@@ -0,0 +1,120 @@
+// +groupName=observabilityui.rhobs
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins,verbs=list;get;watch
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins/status;observabilityuiplugins/finalizers,verbs=get;update
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ObservabilityUIPlugin defines an observability console plugin
+// +k8s:openapi-gen=true
+// +kubebuilder:resource
+// +kubebuilder:subresource:status
+type ObservabilityUIPlugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ObservabilityUIPluginSpec   `json:"spec,omitempty"`
+	Status ObservabilityUIPluginStatus `json:"status,omitempty"`
+}
+
+// ObservabilityUIPluginList contains a list of ObservabilityUIPlugin
+// +kubebuilder:resource
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ObservabilityUIPluginList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ObservabilityUIPlugin `json:"items"`
+}
+
+type UIPluginType string
+
+const (
+	TypeDashboards UIPluginType = "dashboards"
+)
+
+// Observabilityuipluginpec is the specification for desired state of ObservabilityUIPlugin.
+type ObservabilityUIPluginSpec struct {
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=dashboards
+	Type UIPluginType `json:"type"`
+}
+
+// Observabilityuiplugintatus defines the observed state of ObservabilityUIPlugin.
+// It should always be reconstructable from the state of the cluster and/or outside world.
+type ObservabilityUIPluginStatus struct {
+	// Conditions provide status information about the plugin.
+	// +listType=atomic
+	Conditions []Condition `json:"conditions"`
+}
+
+type ConditionStatus string
+
+// +required
+// +kubebuilder:validation:Required
+// +kubebuilder:validation:Pattern=`^([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*/)?(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])$`
+// +kubebuilder:validation:MaxLength=316
+type ConditionType string
+
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+
+	ReconciledCondition        ConditionType = "Reconciled"
+	AvailableCondition         ConditionType = "Available"
+	ResourceDiscoveryCondition ConditionType = "ResourceDiscovery"
+)
+
+type Condition struct {
```

Comment:
I am wondering whether it would make sense to share the `Condition` type between the APIs (same for the https://github.com/rhobs/observability-operator/pull/417). Hopefully, this could allow us to "standardize" the way status is defined and calculated (for each stack/plugin)....are there any downsides (of having one `Condition` type)?